components/formats-bsd/src/loci/formats/out/OMETiffWriter.java

```diff
@@ -106,13 +113,30 @@ public void close() throws IOException {
           populateImage(omeMeta, series);
         }
 
+        String companion = getCompanion();
+        String companionUUID = null;
+        if (null != companion) {
+          String companionXML = getOMEXML(companion);
+          PrintWriter out = new PrintWriter(companion, Constants.ENCODING);
+          out.println(XMLTools.indentXML(companionXML, true));
+          out.close();
+          companionUUID = "urn:uuid:" + getUUID(
+              new Location(companion).getName());
+        }
+
```

Comment:
Refactoring the companion writing logic into a separate method sounds good. However, that cannot be moved inside the loop, since the goal is to write a single companion file linked to all ome-tiffs (e.g. as in @sbesson's example above).